pkg/didcomm/protocol/exchange/service/service.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package service
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/hyperledger/aries-framework-go/pkg/internal/common/support"
+
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/transport"
+
+	"github.com/go-openapi/runtime/middleware/denco"
+
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/api"
+
+	"github.com/hyperledger/aries-framework-go/pkg/common/log"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/exchange"
+)
+
+var logger = log.New("aries-framework/exchange")
+
+// provider contains dependencies for the Exchange protocol and is typically created by using aries.Context()
+type provider interface {
+	OutboundTransport() transport.OutboundTransport
```

Comment:
I don't see the usage of this yet in this PR. If this is a placeholder, how this will be used ?

Reply:
With which packages would we have this import cycle?

Reply:
Context call all services to get list of handlers those services provides.
So services should not refer contexts, it will cause import cycles issue

Reply:
I'll merge to unblock people, but we should check that this conversation got resolved.